### PR TITLE
Correct equations for q-range in sw_instrument

### PR DESCRIPTION
**Pull request by mducle**
Original author: J. R. Stewart (ISIS).

This was contributed by a user who found that the equations used in `sw_instrument` for the kinematic limits were incorrect. That is the equations were correct mathematically but does not actually represent what happens in instruments. Rather they assumed that the detectors extended from `thetamin` to `180-thetamin`. Instead all spectrometers have a much more limited range of scattering angles defined by `thetamin` and a `thetamax` which is signifcantly less than 180 degrees (around 135 degrees maximum for the largest spectrometers; around 60 degrees for smaller / high resolution spectrometers).

This PR corrects this.

----

To Test:

Run an example powder script (e.g. [tutorial 1](https://www.spinw.org/tutorials/01tutorial#7), [5](https://www.spinw.org/tutorials/05tutorial#5), or [6](https://www.spinw.org/tutorials/06tutorial#5)), and run it through `sw_instrument` and check that the kinematic limits look like those in [Mantid PyChop](https://docs.mantidproject.org/nightly/interfaces/direct/PyChop.html)

### CODE REVIEW

**Comment by RichardWaiteSTFC** [comment]
@mducle - is this ready to test? It's been marked draft for a while

**Comment by mducle** [comment]
@RichardWaiteSTFC ah yeah... I meant to write some tests for this but it dropped down the todo list and I forgot about it...

File: swfiles/sw_instrument.m [hunk 1]
```diff
@@ -274,27 +274,33 @@
...
+                Emin = Q*0;
+                ii = find(Q < k0);
+                Emax(ii) = (k0^2-(k0*cosTMin-sqrt(Q(ii).^2-k0^2*sinTMin^2)).^2) * sw_converter(1,'k','meV');
+                Emin(ii) = (k0^2-(k0*cosTMin+sqrt(Q(ii).^2-k0^2*sinTMin^2)).^2) * sw_converter(1,'k','meV');
+                ii = find(Q > k0);
+                Emax(ii) = (k0^2-(k0*cosTMax+sqrt(Q(ii).^2-k0^2*sinTMax^2)).^2) * sw_converter(1,'k','meV');
+                Emin(ii) = (k0^2-(k0*cosTMax-sqrt(Q(ii).^2-k0^2*sinTMax^2)).^2) * sw_converter(1,'k','meV');
             case 2
                 % fix kf
                 Emax = -(k0^2-(k0*cosT+sqrt(Q.^2-k0^2*sinT^2)).^2) * sw_converter(1,'k','meV');
```
Comment (RichardWaiteSTFC) [inline swfiles/sw_instrument.m:297]: This is nit-picking but can we not use `k0` - I think people will assume `k0` is $k_i$ (though I know it's obvious from the comment and the code) ! Can we just call in `kfix` or similar?

File: swfiles/sw_instrument.m [hunk 2]
```diff
@@ -129,10 +129,10 @@
 
 func0 = @swfunc.gaussfwhm;
 
-inpForm.fname  = {'dE'    'ki'  'Ei'  'kf'  'Ef'  'plot' 'polDeg' 'thetaMin'};
-inpForm.defval = {[]      0     0     0     0     false   5        0        };
-inpForm.size   = {[-1 -2] [1 1] [1 1] [1 1] [1 1] [1 1]  [1 1]    [1 1]     };
-inpForm.soft   = {true    false false false false false  false    false     };
+inpForm.fname  = {'dE'    'ki'  'Ei'  'kf'  'Ef'  'plot' 'polDeg' 'thetaMin' 'thetaMax'};
```
Comment (RichardWaiteSTFC) [inline swfiles/sw_instrument.m:132]: Think the new parameter `thetaMax`  needs a do string well

**Review by RichardWaiteSTFC — CHANGES_REQUESTED** [review]
Results agrees well with pychop - can you just add a release note and a doc string, thanks!

**Review by RichardWaiteSTFC — APPROVED** [review]

**Pull request merged**